src/nemos/basis/_transformer_basis.py

```diff
@@ -82,6 +82,11 @@ class TransformerBasis:
     )
 
     def __init__(self, basis: Basis):
+        if not hasattr(basis, "get_params") or not hasattr(basis, "compute_features"):
+            raise TypeError(
+                "Provide an object of type `Basis` to initialize a TransformerBasis. "
+                f"Object of type {type(basis)} provided instead!"
```

Comment:
this error message is confusing -- it's not actually the type that's the issue but the attributes. I'd say that directly (more helpful for someone trying to write their own), mentioning that all nemos basis objects satisfy this criterion